src/v/archival/ntp_archiver_service.cc

```diff
@@ -146,7 +158,12 @@ ss::future<> ntp_archiver::outer_sync_manifest_loop() {
 
     while (!_as.abort_requested()) {
         if (!_parent.is_elected_leader()) {
-            co_await ss::sleep_abortable(_conf->reconciliation_interval, _as);
+            try {
+                co_await _leader_cond.wait();
+            } catch (const ss::broken_condition_variable&) {
+                // stop() was called
+                break;
+            }
             continue;
```

Comment:
The `continue` is not required anymore. We know we are the leader, so we can proceed and start the loop.

Reply:
Fixed.